web/modules/custom/ai_screening_reports/src/Controller/AiScreeningReportsController.php

```diff
@@ -0,0 +1,135 @@
+    private readonly ProjectHelper $projectHelper,
+  ) {
+  }
+
+  /**
+   * Builds the response for displaying project report.
+   */
+  public function __invoke(NodeInterface $node, Request $request): array {
+    return [
+      '#theme' => 'reports_project',
+      '#data' => [
+        'node' => $node,
+        'request' => $request,
+        'projectHelper' => $this->projectHelper,
+        'projectTrackHelper' => $this->projectTrackHelper,
+      ],
+    ];
+  }
+
+  /**
+   * Builds the response for displaying project track chart.
+   */
+  public function projectTrack(Request $request): array|RedirectResponse {
+    $projectTrackIds = $request->get('project_track_id');
+    $projectTracks = [];
+    $loopCounter = 0;
+    // Ensure proper url parameters: ?project_track_id[]=1&project_track_id[]=3.
+    if (is_array($projectTrackIds)) {
+      $projectData = [
+        // @todo get thresholds.
+        'thresholds' => [
+          'x' => 25,
+          'y' => 25,
+          'z' => 25,
+        ],
+        // @todo get possible max the track.
+        'axisMax' => [
+          'x' => 40,
+          'y' => 40,
+          'z' => 40,
+        ],
+        // @todo get labels for axis.
+        'labels' => [
+          'x' => 'XLabel approved threshold',
+          'y' => 'YLabel approved threshold',
+          'z' => 'ZLabel approved threshold',
+        ],
+      ];
+
+      // Create a dataset for each project track.
+      foreach ($projectTrackIds as $id => $projectTrackId) {
+        if (is_numeric($projectTrackId)) {
+          $projectTrack = $this->projectTrackHelper->loadTrack($projectTrackId);
+          if ($projectTrack instanceof ProjectTrackInterface) {
+            $projectTracks[] = $projectTrack;
+            $projectData['dataset'][$id]['chart'] = [
+              'label' => $projectTrack->getProject()->label(),
+              'color' => self::COLOR_CODES[$loopCounter],
+            ];
+            $projectData['dataset'][$id]['plots'] = [
+              // @todo get plots from the track.
+              ['x' => 17, 'y' => 15, 'r' => 3],
+            ];
+            // Set a limit for the number of tracks to display.
+            $loopCounter++;
+            if ($loopCounter >= 15) {
+              $this->messenger()->addWarning($this->t('A maximum of 15 tracks can be displayed.'));
+              break;
+            }
+          }
+        }
+      }
+
+      return [
+        '#theme' => 'reports_project_track',
+        '#attached' => [
+          'drupalSettings' => [
+            'reports_project_track' => $projectData,
+          ],
+        ],
+        '#data' => [
+          'request' => $request,
+          'projectTracks' => $projectTracks,
+        ],
+      ];
+    }
+
+    // If no project track ids could be identified from url params.
+    $this->messenger()->addError($this->t('Access denied. Incorrect url parameters.'));
+    return new RedirectResponse(Url::fromRoute('system.403')->toString());
+  }
```

Comment:
We should report a “bad request” in this case:

```suggestion
    // If no project track ids could be identified from url params.
    throw new BadRequestHttpException();
```

Alternatively, we could set the message and return to the project list page.